pkg/avro/avro.go

```diff
@@ -0,0 +1,71 @@
+package avro
+
+import "encoding/binary"
+
+// AvroCodec implements the Encoder/Decoder interfaces for
+// avro formats
+type AvroCodec struct {
+	encodeSchemaID int
+	schemaCache    *SchemaCache
+}
+
+func NewAvroCodec(schemaID int, cache *SchemaCache) *AvroCodec {
```

Comment:
we can move the avro codec itself to pkg/codec imho. same for proto. schema registry related stuff can stay here.